lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -888,3 +900,45 @@ class _EditableSizeAndTransform {
   final double height;
   final Float64List transform;
 }
+
+/// Used for detecting the changes in the selection.
+///
+/// Currently only used in Firefox.
+///
+/// In Firefox, when cursor moves, nor selectionChange neither onInput
+/// events are triggered. We are listening to keyup event. Selection start,
+/// end values are used to decide if the cursor moved.
+///
+/// Specific keycodes are not checked since users/applicatins can bind their own
+/// keys to move the cursor.
+class SelectionChangeDetection {
+  int _start = 0;
+  int _end = 0;
+
+  /// Method which decides if the selection has changed (cursor moved) compared
+  /// to the previous values.
+  ///
+  /// After each keyup, the start/end values of the selection is compared to the
+  /// previously saved start/end values.
+  bool hasChangeDetected(html.HtmlElement domElement) {
+    if (domElement is html.InputElement) {
+      html.InputElement element = domElement;
+      return _compareSelection(element.selectionStart, element.selectionEnd);
+    }
+    if (domElement is html.TextAreaElement) {
+      html.TextAreaElement element = domElement;
+      return _compareSelection(element.selectionStart, element.selectionEnd);
+    }
+    throw UnsupportedError('Unsupported input type');
+  }
+
+  bool _compareSelection(int selectionStart, int selectionEnd) {
```

Comment:
This method should be named something else because it has a side-effect. Maybe _compareAndCacheSelection? I'd actually prefer that it doesn't have a side effect, and move the caching to the call site.

Reply:
Thanks for the suggestion. I don't like 'and' in method names. I added another method to make it more readable.